static/js/util/profile_edit.js

```diff
@@ -81,3 +104,44 @@ export function saveAndContinue(next) {
     browserHistory.push(next)
   });
 }
+
+export function editProfileObjectArray (arrayName, blankEntry, formCallback) {
+  const { updateProfile, profile } = this.props;
+  if ( profile[arrayName] ) {
+    let editForms = profile[arrayName].map( (entry, index) => (
+      formCallback.bind(this)(entry, index)
+    ));
+    let addAnotherBlankEntry = () => {
+      let clone = Object.assign({}, profile);
+      clone[arrayName].push(Object.assign({}, blankEntry));
+      updateProfile(clone);
+    }
+    editForms.push(
+      <Grid className="profile-tab-grid" key={arrayName}>
+        <Cell col={12} align='middle'>
+          <Button onClick={addAnotherBlankEntry}>
+            Add another entry
+          </Button>
+        </Cell>
+      </Grid>
+    );
+    return editForms;
+  } else {
+    let startEditing = () => {
+      let clone = Object.assign({}, profile);
+      if ( clone[arrayName] === null ) {
+        clone[arrayName] = [ Object.assign({}, blankEntry) ];
```

Comment:
I actually ran into issues with edits being mirrored across objects if I took out `Object.assign` here, so I put it back in. Not exactly sure what that's happening, because looking at the reducer it looks like it shouldn't, so ¯\_(ツ)_/¯